### PR TITLE
docker_* modules: unify docker module version checks

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Allows to specify minimum docker-py version and minimum docker API versions to `AnsibleDockerClient`. (See discussion [here](https://github.com/ansible/ansible/pull/46797#discussion_r224704458).)

Simplifies/superseeds #46527, #46797, helps #38792.

##### ISSUE TYPE
- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/docker_common.py
docker_container
docker_image
docker_image_facts
docker_login
docker_network
docker_secret
docker_service
docker_swarm
docker_swarm_service
docker_volume

##### ANSIBLE VERSION
```
2.8.0
```

### CODE REVIEW

**Review by jwitko — APPROVED** [review]

**Comment by felixfontein** [comment]
bot_status

**Comment by felixfontein** [comment]
@jwitko You have to use `shipit` or `LGTM` to have any effect :)

@dariko @kassiansun @tbouvet Can you take a look here? This affects all `docker_*` modules.

**Comment by jwitko** [comment]
@felixfontein Thanks!  still learning the bot  :)

**Comment by jwitko** [comment]
shipit

File: lib/ansible/module_utils/docker_common.py
Comment (dariko) [inline lib/ansible/module_utils/docker_common.py:186]: Shouldn't `NEEDS_DOCKER_PY2` be `True` if the required version is `>=2.0` instead than `<2.0`?

```suggestion
        NEEDS_DOCKER_PY2 = (LooseVersion(min_docker_version) >= LooseVersion('2.0'))
```

File: lib/ansible/module_utils/docker_common.py
Comment (dariko) [inline lib/ansible/module_utils/docker_common.py:198]: Currently the module would suggest installing `docker` when `min_docker_version < LooseVersion(2.0)` and `docker` or `docker-py` otherwise.
These messages should be swapped, and would be by applying the suggestion above

**Review by kassiansun — APPROVED** [review]

**Comment by kassiansun** [comment]
shipit

**Review by gundalow — APPROVED** [review]
I like `min_docker_api_version`

**Comment by gundalow** [comment]
Merged into `devel` for release in Ansible 2.8.

If you'd like this is older versions please raise a backport PR with changelog/fragment

**Comment by felixfontein** [comment]
@jwitko @kassiansun thanks for your reviews!
@gundalow thanks for merging!

I'll create backport PRs.

**Comment by dariko** [comment]
@felixfontein did you give a look to the review I added in this PR?
Am I failing to understand how this works, or are we committing code with errors?

**Comment by felixfontein** [comment]
@dariko You didn't add a review to this PR AFAIK. Did you maybe forgot to send it? It doesn't show up in the comments here, and also you don't show up in the "reviewers" list at the top right.

**Comment by felixfontein** [comment]
@dariko Could you please explain what you think is a problem? I don't see what could be a problem with this PR.

File: lib/ansible/module_utils/docker_common.py
Comment (felixfontein) [reply lib/ansible/module_utils/docker_common.py:186]: Yes, you are right! Thanks for spotting this. I'll create a new PR for fixing this (since this was already merged).

**Comment by dariko** [comment]
@felixfontein I totally did miss the big green `submit review` button, sorry. I'm not accustomed to github yet.

The problem itself is explained in the review (if my brain didn't imagine it too! :worried:), do you see it now that I submitted it?
In short: I think the test used to define the `NEEDS_DOCKER_PY2` variable should be reversed

**Comment by felixfontein** [comment]
@dariko No problem :) And thanks for noticing this, I think you're totally right!

**Comment by felixfontein** [comment]
(I'll create backport PRs once #47274 is merged, they will then include both PRs.)

**Comment by dariko** [comment]
@felixfontein it's all good, I just thought I was losing it :smile:

**Comment by mohag** [comment]
The docker_swarm module seems to have been working fine with API version 1.30 before this fix. (It does not seem [like there were any swarm-related API changes between 1.30 and 1.35](https://docs.docker.com/engine/api/version-history/), I'm not sure how the API version 1.35 minimum requirement was obtained...) (I'm not using all of its functionality either...)

This breaks the docker_swarm module with the docker-ee version (17.06) required for UCP <= 3.0.x. It worked with Ansible 2.7.0.

(I am aware that 1.35 has always been the documented minimum API version)

**Comment by felixfontein** [comment]
@mohag I created a new issue for this (#48657). Let's continue the discussion there.

**Pull request merged**
